website/content/docs/concepts/integrated-storage.mdx

```diff
@@ -169,15 +179,117 @@ node2    node2.vault.local:8201    follower    true
+commands specifying the explicit names or IPs of the nodes to join to.  In this
+section we look at other TLS-compatible options that lend themselves more to automation.
+
+#### Autojoin with TLS servername
+
+As of Vault 1.6.2, the simplest option might be to specify a leader_tls_servername
+in the retry_join stanza which matches a DNS SAN in the certificate.
+
+Note that names in a certificate's DNS SAN don't actually have to be registered
+in a DNS server.  Your nodes may have no names found in DNS, while still
+using certificate(s) that contain this shared "servername" in their DNS SANs.
+
+#### Autojoin but constrain CIDR, list all possible IPs in certificate
+
+If all the vault node IPs are assigned from a small subnet, e.g. a /28, it
+becomes practical to put all the IPs that exist in that subnet into the IP SANs
+of the TLS certificate the nodes will share.
+
+The drawback here is that the cluster may someday outgrow the CIDR and changing
+it may be a pain.  For similar reasons this solution may be impractical when
+using non-voting nodes and dynamically scaling clusters.
+
+#### Autojoin using a second TCP listener
+
+Modify the Vault HCL to specify a second listener on another port that's used
+solely for auto-join.  This second listener could use disable_tls in conjunction
+with a network ACL or security group that limits access to the Vault nodes.
+
+This technique was recommended at one point but is inferior to the others listed
+here in that it provides a new attack vector.
+
+#### Load balancer instead of autojoin
+
+Most Vault instances are going to have a load balancer (LB) between clients and
+the Vault nodes.  In that case, the LB knows how to route traffic to working
+Vault nodes, and there's no need for auto-join: we can just use retry_join
+with the LB address as the target.
+
+One potential issue here: some users want a public facing LB for clients to
+connect to Vault, but aren't comfortable with Vault internal traffic
+egressing from the internal network it normally runs on.
+
+#### DNS instead of autojoin
+
+One proposal for making autojoin work in AWS is to have Vault reverse resolve
+the IP addresses coming from go-discover and join to the resulting hostname
+instead.  However, if the user is willing and able to configure DNS for each
+Vault node (or is using Consul service registration), we can use DNS instead
+of autojoin, as with the LB case.  In this case the retry_join target address
+becomes the A record that maps to all Vault nodes.
+
+Downsides: Not everyone uses DNS.  Not known to have been tested.  If not using
+Consul for DNS, it feels like it could be slow, because we'd presumably be
+round-robining across all nodes, and at the outset it may take time before we
+actually reach a node that can be joined with.  Presumably in the LB and Consul
+cases health checks would ensure that we don't get routed to a node that can't
+be joined with, those we should validate that.
```

Comment:
Should we either validate this or remove it for now?

Reply:
Removed.